test/common/http/dispatcher_test.cc

```diff
@@ -519,13 +520,123 @@ TEST_F(DispatcherTest, BasicStream) {
   ASSERT_EQ(cc.on_complete_calls, 1);
 }
 
+TEST_F(DispatcherTest, BasicStreamWithTrailers) {
```

Comment:
This test feels quite large and it's kind of difficult to understand what suite of behaviors we're trying to exercise here. Do you know if we can isolate the specific behaviors and have smaller individual tests?

Reply:
Some of these tests can be condensed a little bit by taking the repeated parts into utility functions in the test class. I was thinking about this as I was adding more tests. Will do in a subsequent PR https://github.com/lyft/envoy-mobile/issues/794

Reply:
I think that's great to do also (I am slightly opinionated about test utils but we can talk about that separately)

The confusion I have is more on the behavior exercised by this test (since it seems to be asserting on all the different handlers). Does this test primarily exercise a general end to end behavior of a basic stream?

Reply:
yeah, it is an http stream with trailers. There are individual tests for a headers only request, one with a body, and now this one with trailers.